### PR TITLE
Update the call order of `afterSSR()` actions in `frontity/core`

### PR DESCRIPTION
**Pull request by michalczaplinski**
**What**:

Update the call order of `afterSSR()` actions in `frontity/core`

**Why**:

We need to call `afterSSR()` **before** taking the state snapshot so that the action can still modify the state before that snapshot is taken.

**Tasks**:

- [x] Code
- [x] TSDocs
- [x] TypeScript
- [x] End to end tests
- [x] Add a changeset (with link to its [Feature Discussion](https://community.frontity.org/c/33) if it exists)

**Unrelated Tasks**

- [ ] Unit tests
- [ ] TypeScript tests
- [ ] Update starter themes
- [ ] Update other packages
- [ ] Update community discussions

### CODE REVIEW

**Comment by luisherranz** [comment]
I've run the e2e tests again to see if they pass the infinite scroll now.

It seems like the infinite scroll tests are not very reliable. Maybe the scroll support of Cypress is not good enough?

File: packages/core/src/server/middlewares/server-side-rendering.ts [hunk 1]
```diff
@@ -67,6 +84,11 @@ export const serverSideRendering = async (
     // https://github.com/smooth-code/loadable-components/pull/239#issuecomment-482501467
     const customExtractor = extractor as Extractor;
 
+    // Run afterSSR actions. It runs at this point because we want to run it
+    // before taking the state snapshot. This gives the user a chance to
+    // modify the state before sending it to the client
+    runAfterSSRActions(store, ctx);
```
Comment (cristianbote) [inline packages/core/src/server/middlewares/server-side-rendering.ts]: Do we have to `await` here? 🤔  I think so? Same thing in the other case.
```suggestion
    await runAfterSSRActions(store, ctx);
```

**Comment by luisherranz** [comment]
> I've run the e2e tests again to see if they pass the infinite scroll now

They passed this time 🤷

File: packages/core/src/server/middlewares/server-side-rendering.ts (hunk 1 above)
Comment (michalczaplinski) [reply packages/core/src/server/middlewares/server-side-rendering.ts]: Oh, you're right! For some reason I thought it should not be necessary to await here if I await **inside** each of the `afterSSR()` calls but that's not the case of course.

**Comment by michalczaplinski** [comment]
@cristianbote Updated to `await runAfterSSRActions()` like you pointed out. Feel free to merge after review :)

**Comment by cristianbote** [comment]
> > I've run the e2e tests again to see if they pass the infinite scroll now
> 
> They passed this time 🤷

And then failed next time they ran. 🤔  Triggered them again.

**Review by cristianbote — APPROVED** [review]

**Pull request merged**